### PR TITLE
Update create actor delegation handler

### PR DESCRIPTION
**Pull request by defectiveAi**

### CODE REVIEW

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Commands/Delegations/CreateMessageDelegationDto.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 namespace Energinet.DataHub.MarketParticipant.Application.Commands.Delegations
 {
-    public sealed record CreateActorDelegationDto(
+    public sealed record CreateMessageDelegationDto(
```
Comment (michaelmajgaard) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Commands/Delegations/CreateMessageDelegationDto.cs:21]: Should these be typed to our `ActorId` and `GridAreaId`? All properties in `MessageDelegationDto` are typed.

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Delegations/CreateMessageDelegationHandler.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            var actorDelegatedTo = await actorRepository
+                .GetAsync(new ActorId(request.CreateDelegation.DelegatedTo))
+                .ConfigureAwait(false);
+
+            NotFoundValidationException.ThrowIfNull(actor, request.CreateDelegation.DelegatedFrom);
+            NotFoundValidationException.ThrowIfNull(actorDelegatedTo, request.CreateDelegation.DelegatedTo);
+
+            if (actor.Status != ActorStatus.Active || actorDelegatedTo.Status != ActorStatus.Active)
+            {
+                throw new ValidationException("Actors to delegate from/to must be both be active to delegate messages.")
```
Comment (michaelmajgaard) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Delegations/CreateMessageDelegationHandler.cs:54]: Remove the first `be` in the message.

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Commands/Delegations/CreateMessageDelegationDto.cs [hunk 3]
```diff
@@ -18,7 +18,7 @@
...
         Guid DelegatedFrom,
```
Comment (michaelmajgaard) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Commands/Delegations/CreateMessageDelegationDto.cs:22]: StopsAt not possible when creating

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Commands/Delegations/CreateMessageDelegationDto.cs (hunk 1 above)
Comment (FirestarJes) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Commands/Delegations/CreateMessageDelegationDto.cs:21]: Fixed

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Delegations/CreateMessageDelegationHandler.cs (hunk 2 above)
Comment (FirestarJes) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Delegations/CreateMessageDelegationHandler.cs:54]: Fixed

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Delegations/CreateMessageDelegationHandler.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                    .WithErrorCode("message_delegation.actors_from_and_to_inactive");
```
[the remark this answers is not in the document]
Comment (FirestarJes) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Delegations/CreateMessageDelegationHandler.cs:55]: Fixed

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Commands/Delegations/CreateMessageDelegationDto.cs (hunk 3 above)
Comment (FirestarJes) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Commands/Delegations/CreateMessageDelegationDto.cs:22]: Fixed

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Delegations/GetDelegationsForActorHandler.cs [hunk 5]
```diff
@@ -24,26 +25,46 @@
...
                 .ConfigureAwait(false);
 
             NotFoundValidationException.ThrowIfNull(actor, request.ActorId);
 
-            return new GetDelegationsForActorResponse(new Collection<ActorDelegationDto>());
+            var delegations = await messageDelegationRepository
+                .GetForActorAsync(new ActorId(request.ActorId))
+                .ConfigureAwait(false);
+
+            var result = new List<MessageDelegationDto>();
```
Comment (michaelmajgaard) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Delegations/GetDelegationsForActorHandler.cs:47]: Could skip creating af list here and just `.Select(..)` on `delegations` and defer enumeration for later. `GetDelegationsForActorResponse` should take an `IEnumerable` as input.

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs [hunk 6]
```diff
@@ -83,7 +83,7 @@ public void DelegateTo(ActorId delegatedTo, GridAreaId gridAreaId, Instant start
         _delegations.Add(delegationPeriod);
     }
 
-    public void StopDelegation(DelegationPeriod existingPeriod, Instant stopsAt)
+    public void StopDelegation(DelegationPeriod existingPeriod, Instant? stopsAt)
```
Comment (michaelmajgaard) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs:86]: Does it make sense to have nullable `stopsAt`?

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Commands/Delegations/StopMessageDelegationDto.cs [hunk 7]
```diff
@@ -17,7 +17,8 @@
 
 namespace Energinet.DataHub.MarketParticipant.Application.Commands.Delegations
 {
-    public sealed record UpdateActorDelegationDto(
-        ActorDelegationId Id,
-        DateTimeOffset ExpiresAt);
+    public sealed record StopMessageDelegationDto(
+        MessageDelegationId Id,
+        DelegationPeriodId PeriodId,
+        DateTimeOffset? StopsAt);
```
Comment (michaelmajgaard) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Commands/Delegations/StopMessageDelegationDto.cs:23]: Makes sense to be nullable?

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs (hunk 6 above)
Comment (FirestarJes) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs:86]: Yes, as they use it to say that it should run indefinitely

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Commands/Delegations/StopMessageDelegationDto.cs (hunk 7 above)
Comment (FirestarJes) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Commands/Delegations/StopMessageDelegationDto.cs:23]: As above

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs (hunk 6 above)
Comment (michaelmajgaard) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs:86]: Tjek 👍

**Review by michaelmajgaard — APPROVED** [review]

**Pull request merged**